pandas/tests/scalar/interval/test_arithmetic.py

```diff
@@ -0,0 +1,54 @@
+from datetime import timedelta
+
+import numpy as np
+import pytest
+
+from pandas import Interval, Timedelta, Timestamp
+
+
+@pytest.mark.parametrize("method", ["__add__", "__sub__"])
+@pytest.mark.parametrize(
+    "delta", [Timedelta(days=7), timedelta(7), np.timedelta64(7, "D")]
+)
+def test_timestamp_interval_add_subtract_timedelta(method, delta):
+    # https://github.com/pandas-dev/pandas/issues/32023
+    interval = Interval(
+        Timestamp("2017-01-01 00:00:00"), Timestamp("2018-01-01 00:00:00")
+    )
+
+    result = getattr(interval, method)(delta)
+
+    left = getattr(interval.left, method)(delta)
+    right = getattr(interval.right, method)(delta)
+    expected = Interval(left, right)
+
+    assert result == expected
+
+
+@pytest.mark.parametrize(
+    "delta", [Timedelta(days=7), timedelta(7), np.timedelta64(7, "D")]
+)
+def test_timedelta_add_timestamp_interval(delta):
```

Comment:
this would be somewhat clearer as `test_timestamp_interval_radd_timedelta` (or just rolled into the previous test)